spec/order_list_spec.rb

```diff
@@ -0,0 +1,49 @@
+require 'order_list'
+
+describe OrderList do
+
+  let(:pizza){double(:pizza, name: :pizza, price: 1)}
+  let(:chicken){double(:chicken, name: :chicken, price: 2)}
+  let(:dish_list){double(:dish_list, price_list: {pizza: 1, chicken: 2}, dishes: [pizza, chicken])}
+  subject(:order_list) {described_class.new(dish_list)}
+  before(:example){allow(order_list.client.messages).to receive(:create)}
+
+  describe '#price_list' do
+    it 'gives a list with the price of each dish' do
+      expect(dish_list).to receive(:price_list)
+      order_list.price_list
+    end
+  end
+
+  describe '#place_order' do
+
+    context 'a dish selected is unavailable' do
+      it 'raises an error' do
+        expect{order_list.place_order(apple: 2)}.to raise_error "An ordered dish is unavailable"
+      end
+    end
+
+    context 'all dishes selected are available' do
+      it 'does not raise an error' do
+        expect{order_list.place_order(pizza: 1)}.not_to raise_error
+      end
+
+      context 'when given total does not match sum of dishes in customer\'s order' do
```

Comment:
Metrics/LineLength: Line is too long. [85/80] (https://github.com/bbatsov/ruby-style-guide#80-character-limits)